### ISSUE TITLE
yq doesn't merge when there is a reference

### ISSUE THREAD

**Issue by snedelchev** [mikefarah/yq#374]
**yq doesn't merge when there is a reference**
Doesn't merge 2 files when there is a reference.

**Input Yaml**
data1.yml:
```common: &common
  in_parallel:
    limit: 4
  resource:
    check_every: 5m
# Reuse all the values from `common`
dev-cloud: *common

```

data2.yml:
```
dev-cloud: # NOTE: this inherits from `*common`
  clusters:
    dev:
      kops-cluster-name: dev-k8s.example.com
      kubernetes-namespace: dev

```

**Command**
The command you ran:
```
yq merge data1.yml data2.yml
```

**Actual behavior**

```common: &common
  in_parallel:
    limit: 4
  resource:
    check_every: 5m
# Reuse all the values from `common`
dev-cloud: *common
```

**Expected behavior**

```common:
  in_parallel:
    limit: 4
  resource:
    check_every: 5m
dev-cloud:
  clusters:
    dev:
      kops-cluster-name: dev-k8s.example.com
      kubernetes-namespace: dev
  in_parallel:
    limit: 4
  resource:
    check_every: 5m

```

**it's works fine with version 2.4.1**

**Comment by mikefarah** [comment]
This is a little hard to follow - can you provide a simple example including the 'common' anchor? What do you mean in data2 by '# NOTE: this inherits from `*common`' ?
